### ISSUE TITLE
As a user I can authenticate with Login.gov

### ISSUE THREAD

**Issue by Jkrzy** [GSA-TTS/FAC#135]
**Background**
We're using Login.gov to authenticate

**Stories**

1. As an Auditee or Auditor I know how and where I need to authenticate so that I may submit new and modify existing single audit packages.

**Acceptance Criteria (We'll know we're done when..._**
- [ ] When a user can authenticate with login.gov and be associated with a Django user account
- [ ] For a user who has authenticated with login.gov and does not yet have a Django user account, the application creates one
  
**Tasks**
- [x] https://github.com/GSA-TTS/FAC/issues/259
- [ ] https://github.com/GSA-TTS/FAC/issues/267
- [x] https://github.com/GSA-TTS/FAC/issues/268
- [x] https://github.com/GSA-TTS/FAC/issues/308
- [x] Make contact with login.gov partnerships team

**QA**
- [ ] Meets acceptance criteria
- [ ] Screen reader - Listen to the experience with a screen reader extension, ensure the information presented in order
- [ ] Keyboard navigation - Run through acceptance criteria with keyboard tabs, ensure it works. 
- [ ] Text scaling - Adjust viewport to 1280 pixels wide and zoom to 200%, ensure everything renders as expected. Document 400% zoom issues with USWDS if appropriate.

**Comment by MelissaBraxton** [comment]
Q relevant to log in and where we decide to implement the login button on the front end - 

In the current FAC, all **data submission** and **getting submitted data** tasks require logging in. Will that need to be the case with our system? Or is it possible we could make it so that folks wont need to sign in to get the data?

**Comment by jadudm** [comment]
Thoughts, mostly.

*Obtaining* data from the FAC... currently, you don't need to log in. That would represent a change from the existing system. I can visit [https://facdissem.census.gov/](https://facdissem.census.gov/) and search without logging in.

So, if the question is "can a member of the public obtain FAC data without logging in," the current system says **yes**. If our workflow currently suggests otherwise, then we probably need to pause on that and ask *why*? My suspicion/starting assumption is that we should not put barriers between the public and this data. 

The current mocks suggest there is a login button on every page, suggesting it might be part of a site-wide template. (Given where we're at, "suggesting" might be too strong a word.) If it was, it could potentially communicate to users that they will, somehow, need to log in to use the site -- which is not true.

**Comment by MelissaBraxton** [comment]
Apologies! I think I was confusing sam.gov login requirement to get data w/ the FAC. You're right that folks dont log in to get FAC data. 

Agree that the Login button is not in the right place (it was there on the mocks purely as a research probe), and it's something we need to sort out as part of the Login story, but it's connected to us also getting concrete on the overall IA, which @brentryanjohnson is creating a new issue for.

**Comment by petermarks-gsa** [comment]
I broke out the work @timoballard has already completed. Moving this story / parent issue back to in progress until the last issue (https://github.com/GSA-TTS/FAC/issues/267) is ready for review.

**Comment by petermarks-gsa** [comment]
@heymatthenry @timoballard Can you

**Comment by petermarks-gsa** [comment]
@heymatthenry @timoballard Can you please outline steps to QA this end to end? I tried the [Federalist preview of main](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/site/gsa-tts/fac-frontend/audit/submission/) and got an error: 
<img width="1079" alt="Screen Shot 2022-07-06 at 8 29 15 AM" src="https://user-images.githubusercontent.com/100610443/177587947-5ab3abad-72f4-41db-b589-826008a58690.png">

**Comment by timoballard** [comment]
@petermarks-gsa (cc: @heymatthenry) - this is due to a missing configuration item on the Login.gov side. Specifically, you have to pre-register your post-login redirect URIs with Login (to protect against in-the-middle attacks where you get redirected back to a site other than the one you came from, access token in hand). 

I updated the Login configuration to include the federalist/main preview URI. I can also add you all as team members with access to our Login sandbox app so that you're capable of making these kinds of edits going forward. Will follow up on Slack.

**Comment by petermarks-gsa** [comment]
@timoballard @heymatthenry Here's my end-to-end QA Pass:

* ✅ Clicking "Submit at audit" in nav brings up modal, which can be dismissed
* ❌ Clicking "Log in to submit an audit" on the home page should do same as ^^
* ❌ When I log in with login.gov with a sandbox-approved email (peter.marks@gsa.gov) I am redirected to the first step of the pre-SAC page (❌ remain on home page)
* [blocked on https://github.com/GSA-TTS/FAC/issues/322] I fill out forms in the SAC flow and progress to the next step. 
* [blocked on access to django admin] When I log into django admin, I see a SAC model assigned to my user with the data I entered.

**Comment by timoballard** [comment]
Thanks @petermarks-gsa - I can clean up items 2 and 3. One minor clarification: any sandbox-login account will be able to access the site. The adding of the FAC team's email addresses just allows you all to edit the sandbox-login configuration (if you follow [this link](https://dashboard.int.identitysandbox.gov/service_providers), you should see the `GSA FAC` app listed)

**Comment by petermarks-gsa** [comment]
@timoballard Thanks for clarifying. Are you able to set me up with access to django admin on the [Federalist preview of main](https://federalist-9a617ff8-042d-4076-9581-bb999f9c6639.app.cloud.gov/site/gsa-tts/fac-frontend/audit/submission/)?

**Comment by timoballard** [comment]
@petermarks-gsa I realized this morning that my cloud.gov access isn't _quite_ there (I have access to the org, but can't see the FAC space/app). My bad for not verifying this the first time around. Once this is sorted out I can get you setup with admin access (or @tadhg-ohiggins can probably take care of it if we don't want to wait for my access to get sorted out)

**Comment by timoballard** [comment]
@petermarks-gsa - the issues you found in QA were addressed in [#29](https://github.com/GSA-TTS/FAC-Frontend/pull/29). We'll get your django-admin access in place once our cloud.gov permissions get sorted!
